### DISCUSSION TITLE
Typing for make_response when using timeout

### DISCUSSION THREAD

**Discussion by pamelafox** [pallets/quart#269] [Q&A]
I have this code:

        response = await make_response(format_as_ndjson(response_generator))
        response.timeout = None

The make_response function returns either quart's Response or Werkzeug's Response. Only quart's Response has a timeout property.

Pyright thusly errors when it sees my code using response.timeout:

  /workspaces/azure-search-openai-demo/app/backend/app.py:126:18 - error: Cannot assign member "timeout" for type "Response"
    Member "timeout" is unknown (reportGeneralTypeIssues)

Should I always check if its an instance of quart's response before setting it, or is there a different approach? The doc examples all set timeout with no checks.

**Comment by pgjones** [comment]
The docs were written before Quart supported Werkzeug's Responses - I'm actually thinking of trimming the docs to only refer to difference to Flask, but this is an aside.

I would either cast or type: ignore in this case - I'm not sure of a better approach at the moment.

**Reply by pamelafox** [reply]
Okay, we'll ignore for now. Should it maybe use generics or overloads? I haven't looked into the Quart code to see when it uses each response type. Thanks for the reply!
